### PR TITLE
Feature: Add and Delete a Comment to a Post

### PR DESCRIPTION
**Pull request by EJamesssss**
Hi @superspike7 ,  sorry siningit ko lang muna yung comments and comment form sa social feed para makita sya ni @reyolan . panget lang sya tingnan sa ngayon dahil sa siningit ko na comments, pero di pa naman kasama sa design mo yung comments no ?

Pareview na rin guys, thank you! 💯

### CODE REVIEW

**Review by reyolan — CHANGES_REQUESTED** [review]
Personally, I intend to avoid using hidden fields since it is editable via DOM wherein a user can just edit the link and he/she can comment posing as another user. (That is a security risk).

See this sample stock application: 
image
Because of hidden fields, the user was able to input an invalid stock and invalid amount (negative) because he/she can edit the DOM via inspect element and in the server-side there is no particular security avoids this scenario (like validations, etc.) 

Imagine that if we can edit the user_id in DOM, then we can comment as another user.

Instead of utilizing hidden fields for user_id and post_id. You can do scoping.

Instead of `Comment.new`. You can do  `current_user.comments.build`. In this way, user_id is integrated immediately when creating a comment.

the route would be like: /posts/:id/comment/new (nested route)
image

So in the create action of the CommentsController, it would be `current_user.comments.build(comment_params)` then in your private method, `comment_params` would then just require the body and we can merge params_id there.
The POST route would be like: /posts/:id/comments.

The the private method would be

image


The important part here is we must not expose the user_id since you can then pose as another user. Changing the post id will just change the ownership of whom the comment belongs to. It is somehow okay but really not desireable.

TL;DR: Only use hidden fields for styling purposes/conveniences or you have a validation in the server-side (back-end) that confirms that the user_id belongs to you.

Note: Not sure if there will be a syntax error in the code that I have shown, cinode ko lang sya on the spot so you may need to do a minor change if there is a syntax error or if creating the comment does not work as intended.

**Comment by EJamesssss** [comment]
> Personally, I intend to avoid using hidden fields since it is editable via DOM wherein a user can just edit the value in the hidden field and he/she can comment posing as another user. (That is a security risk).
> 
> See this sample stock application: image Because of hidden fields, the user was able to input an invalid stock and invalid amount (negative) because he/she can edit the DOM via inspect element and in the server-side there isn't any particular security that avoids this scenario (like validations, etc.)
> 
> Imagine that if we can edit the user_id in DOM. We can then comment as another user.
> 
> Instead of utilizing hidden fields for user_id and post_id. You can do scoping.
> 
> Instead of `Comment.new`. You can do `current_user.comments.build`. In this way, user_id is integrated immediately when creating a comment.
> 
> the route would be like: `/posts/:id/comment/new` (nested route) image
> 
> So in the create action of the CommentsController, it would be `current_user.comments.build(comment_params)` then in your private method, `comment_params` would then just permit the body and we can merge params_id there. The POST route would be like: `/posts/:id/comments`.
> 
> The private method would be
> 
> image
> 
> The comment_params would look like this: image
> 
> The important part here is we must not expose the user_id since you can then pose as another user. Changing the post id will just change the ownership of whom the comment belongs to. It is somehow okay but not really desireable.
> 
> TL;DR: Only use hidden fields for styling purposes/conveniences or if you have a validation in the server-side (back-end) that confirms that the user_id belongs to you. Please change on how the comment is created (do not use hidden fields)
> 
> Note: Not sure if there will be a syntax error in the code that I have shown, cinode ko lang sya on the spot so you may need to do a minor change if there is a syntax error or if creating the comment does not work as intended.
> 
> Extra Note: IF you really need to use hidden field for storing a required attribute in the model, alternatively, you can use sessions hash, basically, it's a way to persist data from action to another action.



> Personally, I intend to avoid using hidden fields since it is editable via DOM wherein a user can just edit the value in the hidden field and he/she can comment posing as another user. (That is a security risk).
> 
> See this sample stock application: image Because of hidden fields, the user was able to input an invalid stock and invalid amount (negative) because he/she can edit the DOM via inspect element and in the server-side there isn't any particular security that avoids this scenario (like validations, etc.)
> 
> Imagine that if we can edit the user_id in DOM. We can then comment as another user.
> 
> Instead of utilizing hidden fields for user_id and post_id. You can do scoping.
> 
> Instead of `Comment.new`. You can do `current_user.comments.build`. In this way, user_id is integrated immediately when creating a comment.
> 
> the route would be like: `/posts/:id/comment/new` (nested route) image
> 
> So in the create action of the CommentsController, it would be `current_user.comments.build(comment_params)` then in your private method, `comment_params` would then just permit the body and we can merge params_id there. The POST route would be like: `/posts/:id/comments`.
> 
> The private method would be
> 
> image
> 
> The comment_params would look like this: image
> 
> The important part here is we must not expose the user_id since you can then pose as another user. Changing the post id will just change the ownership of whom the comment belongs to. It is somehow okay but not really desireable.
> 
> TL;DR: Only use hidden fields for styling purposes/conveniences or if you have a validation in the server-side (back-end) that confirms that the user_id belongs to you. Please change on how the comment is created (do not use hidden fields)
> 
> Note: Not sure if there will be a syntax error in the code that I have shown, cinode ko lang sya on the spot so you may need to do a minor change if there is a syntax error or if creating the comment does not work as intended.
> 
> Extra Note: IF you really need to use hidden field for storing a required attribute in the model, alternatively, you can use sessions hash, basically, it's a way to persist data from action to another action.

Hi @reyolan , please review my refactored comment feature, users can still comment in social media feed and on show post function, the only difference is the hidden fields are remove as you suggested. (Thank you, another learning yung naexplore ko pano makuha yung id as value sa form without using params 💯 )

**Review by reyolan — APPROVED** [review]
Looks good to me, good job!
